### PR TITLE
Firefall Client Error Code Support

### PR DESCRIPTION
**Pull request by roribio**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: actions/FirefallClient.js [hunk 1]
```diff
@@ -42,29 +54,32 @@ class FirefallClient {
           n: 1,
         },
       })
-      .json();
+      .json()
+      .catch((error) => {
+        // console.error('Error generating completion:', error);
```
Comment (vtsaplin) [inline actions/FirefallClient.js:59]: I suggest using a logger here, as this information is required to troubleshoot action errors.

File: actions/FirefallClient.test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+jest.mock('./Network.js');
+
+const firefall = new FirefallClient('endpoint', 'apiKey', 'org', 'accessToken');
+let error;
+
+beforeEach(() => {
+  jest.clearAllMocks();
+
+  error = new WretchError('Internal Server Error');
```
Comment (vtsaplin) [inline actions/FirefallClient.test.js:25]: I would avoid using this dangling mutable variable and replace it with a factory function like this:

```js
function createWretchError(status) {
   const error = new WretchError('Internal Server Error');
   error.status = status;
   return error;
}
```

**Review by vtsaplin — COMMENTED** [review]
Overall, this looks good to me, with just a few minor improvements.
And please do not forget to update the UI to use error.message.

File: actions/FirefallClient.js (hunk 1 above)
Comment (roribio) [reply actions/FirefallClient.js:59]: For this I considered two approaches:

1. Passing the logger object from the "AuthAction" and reusing it in the FirefallClient.
2. Creating a new logger object in the FirefallClient by passing the LOG_LEVEL to the FirefallClient.

I think #1 makes more sense, what do you suggest?

Thanks.

Comment (vtsaplin) [reply actions/FirefallClient.js:59]: The logging has already been implemented. Just merge this one with the main branch: https://github.com/adobe/aem-genai-assistant/pull/195

**Pull request closed without merging**